routes/learner.route.js

```diff
@@ -28,6 +28,40 @@ router.get("/topic/:id/finish", async(req, res) => {
   })
 });
 
+router.post("/topic/:id/finish", async function(req, res) {
+  const topicid = req.params.id
+  const userid = res.locals.authUser.userid
+  const words = req.body.words
+
+  const hasLearned = await learnerService.hasLearnedTopic(userid, topicid)
+  if (hasLearned) {
+    return
+  }
+
+  const finishWords = []
+  const timestamp = new Date()
+  const wordData = {
+    userid: userid,
+    memorylevel: 1,
+    firsttime: timestamp,
+    updatetime: timestamp,
+    isstudy: false,
+  }
+  for (let i = 0; i < words.length; i++) {
+    const wordid = words[i].wordid
+    const word = {wordid, ...wordData}
+    finishWords.push(word)
+  }
```

Comment:
```suggestion
  for (const word_item in words) {
    const wordid = word_item.wordid
    finishWords.push({wordid, ...wordData})
  }
```

Reply:
checkout [map in javascript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)

Reply:
dùng map thì khác gì ô

Reply:
map chạy nhanh hơn vòng for ông, trong code proj thiệt ngta hạn chế dùng for nhất có thể, for chậm lắm